content/docs/instrumenting/exposition_formats.md

```diff
@@ -5,104 +5,96 @@ sort_rank: 6
+| **Limitations** | <ul><li>Verbose</li><li>Types and docstrings not integral part of the syntax, meaning little-to-nonexistent metric contract validation</li><li>Parsing cost</li></ul>|
+| **Supported metric primitives** | <ul><li>Counter</li><li>Gauge</li><li>Histogram</li><li>Summary</li><li>Untyped</li></ul> |
+
 
 ### Text format details
 
-The protocol is line-oriented. A line-feed character (`\n`) separates lines.
-The last line must end with a line-feed character. Empty lines are ignored.
+Prometheus text-based format is line oriented. Lines are separated by a line
+feed character (`\n`). The last line must end with a line feed character.
+Empty lines are ignored.
+
+#### Line format
 
 Within a line, tokens can be separated by any number of blanks and/or tabs (and
-have to be separated by at least one if they would otherwise merge with the
-previous token). Leading and trailing whitespace is ignored.
+must separated by at least one if they would otherwise merge with the previous
+token). Leading and trailing whitespace is ignored.
+
+#### Comments, help text, and type information
 
 Lines with a `#` as the first non-whitespace character are comments. They are
 ignored unless the first token after `#` is either `HELP` or `TYPE`. Those
 lines are treated as follows: If the token is `HELP`, at least one more token
 is expected, which is the metric name. All remaining tokens are considered the
 docstring for that metric name. `HELP` lines may contain any sequence of UTF-8
-characters (after the metric name), but the backslash and the line-feed
+characters (after the metric name), but the backslash and the line feed
 characters have to be escaped as `\\` and `\n`, respectively. Only one `HELP`
-line may exist for the same metric name.
+line may exist for any given metric name.
 
 If the token is `TYPE`, exactly two more tokens are expected. The first is the
 metric name, and the second is either `counter`, `gauge`, `histogram`,
 `summary`, or `untyped`, defining the type for the metric of that name. Only
-one `TYPE` line may exist for the same metric name. The `TYPE` line for a
-metric name has to appear before the first sample is reported for that metric
+one `TYPE` line may exist for a given metric name. The `TYPE` line for a
+metric name must appear before the first sample is reported for that metric
 name. If there is no `TYPE` line for a metric name, the type is set to
 `untyped`. Remaining lines describe samples, one per line, with the following
-syntax (EBNF):
+syntax ([EBNF](https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form)):
+
+```
+metric_name [
+  "{" label_name "=" `"` label_value `"` { "," label_name "=" `"` label_value `"` } [ "," ] "}"
+] value [ timestamp ]
+```
 
-    metric_name [
-      "{" label_name "=" `"` label_value `"` { "," label_name "=" `"` label_value `"` } [ "," ] "}"
-    ] value [ timestamp ]
+`metric_name` and `label_name` have the usual Prometheus expression language restrictions:
```

Comment:
The following bullet points that are introduced here don't actually further describe this sentence, they are completely separate points. So either everything should be a bullet point (with a different intro statement) or nothing should be...

Reply:
Ah, okay. I'm just going to restructure this list.